judge/views/api/api_v1.py

```diff
@@ -38,10 +38,13 @@ def api_v1_contest_detail(request, contest):
         can_see_rankings = False
 
     problems = list(contest.contest_problems.select_related('problem')
-                    .defer('problem__description').order_by('order')) if in_contest else []
+                    .defer('problem__description').order_by('order'))
     users = base_contest_ranking_list(contest, problems, contest.users.filter(virtual=0)
                                       .prefetch_related('user__organizations')
                                       .order_by('-score', 'cumtime')) if can_see_rankings else []
+    if not in_contest and not contest.ended:
```

Comment:
This should be reordered so that the code that deals with the problems is all in one place, and the code that deals with users follows it, rather than being inserted right in its midst.